lib/engine/game/g_1856.rb

```diff
@@ -47,6 +48,96 @@ class G1856 < Base
 
       HOME_TOKEN_TIMING = :operating_round
 
+      def setup
+        @straight_city ||= @tiles.find { |t| t.name == '57' }
+        @sharp_city ||= @tiles.find { |t| t.name == '5' }
+        @gentle_city ||= @tiles.find { |t| t.name == '6' }
+
+        @straight_track ||= @tiles.find { |t| t.name == '9' }
+        @sharp_track ||= @tiles.find { |t| t.name == '7' }
+        @gentle_track ||= @tiles.find { |t| t.name == '8' }
+
+        @x_city ||= @tiles.find { |t| t.name == '14' }
+        @k_city ||= @tiles.find { |t| (t.name == '15') }
+      end
+
+      def gray_phase?
```

Comment:
why do you need this? can't you just do @phase.name == :gray or something?

Reply:
Can lib/engine/step/g_1856 access @phase ? If so this method is not needed

Reply:
If you are asking why have the helper, this keeps the logic inside here, so that if it needs to be optimized later it only gets optimized in one place

If you are asking why `@phase.tiles.include?('gray')`, as an alternative `%w[6 8 D].include?(@phase.name)` can be used but between the two `@phase.tiles.include?('gray')` seems to be more straightforward of the two